### PR TITLE
Use a single enum and switch that replaces no_subsample, force_subsample

### PR DESCRIPTION
**Pull request by Elad-Laufer**

### CODE REVIEW

File: libvips/foreign/vips2jpeg.c [hunk 1]
```diff
@@ -628,18 +628,25 @@ write_vips( Write *write, int qfac, const char *profile,
...
-			write->cinfo.comp_info[i].v_samp_factor = 1;
-		}
+	switch (subsample_mode) {
+        case VIPS_FOREIGN_JPEG_SUBSAMPLE_ON:
+            break;
+        case VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO:
+            /* Turn off chroma subsampling. Follow IM and do it automatically for
+            * high Q.
+            */
+            if( !(no_subsample ||
```
Comment (alon-ne) [inline libvips/foreign/vips2jpeg.c]: Now that we have subsample_mode, it seems that the no_subsample parameter is redundant and also confusing. 
If a user doesn't want subsampling, he wouldn't pass subsample_mode=VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO and no_subsample=1, he should just pass subsample_mode=VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF

File: libvips/include/vips/foreign.h [hunk 2]
```diff
@@ -364,6 +364,18 @@ int vips_vipssave( VipsImage *in, const char *filename, ... )
 int vips_openslideload( const char *filename, VipsImage **out, ... )
 	__attribute__((sentinel));
 
+/**
+ * VipsForeignJpegSubsample:
+ * @VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO: default preset
+ * @VIPS_FOREIGN_JPEG_SUBSAMPLE_ON: always perform subsampling
+ * @VIPS_FOREIGN_JPEG_SUBSAMPLE_ON: never perform subsampling
```
Comment (alon-ne) [inline libvips/include/vips/foreign.h]: Fix comment ON -> OFF

File: test/test-suite/test_foreign.py [hunk 3]
```diff
@@ -259,14 +259,17 @@ def test_jpegsave(self):
         im = pyvips.Image.new_from_file(JPEG_FILE)
 
         # higher Q should mean a bigger buffer
-        b1 = im.jpegsave_buffer(Q=10)
-        b2 = im.jpegsave_buffer(Q=90)
-        assert len(b2) > len(b1)
+        q10 = im.jpegsave_buffer(Q=10)
+        q10o = im.jpegsave_buffer(Q=10, subsample_mode=2)
```
Comment (alon-ne) [inline test/test-suite/test_foreign.py]: Let's rename q10o -> q10_subsample_off

File: test/test-suite/test_foreign.py [hunk 4]
```diff
@@ -259,14 +259,17 @@ def test_jpegsave(self):
...
+        q90 = im.jpegsave_buffer(Q=90)
+        assert len(q90) > len(q10)
+        assert len(q10o) > len(q10)
 
         # force subsampling should result in smaller buffer
-        b1 = im.jpegsave_buffer(Q=90, force_subsample=True)
-        b2 = im.jpegsave_buffer(Q=90)
-        assert len(b2) > len(b1)
+        q90s = im.jpegsave_buffer(Q=90, subsample_mode=1)
```
Comment (alon-ne) [inline test/test-suite/test_foreign.py]: Let's rename q90s -> q90_subsample_on

File: test/test-suite/test_foreign.py [hunk 5]
```diff
@@ -259,14 +259,17 @@ def test_jpegsave(self):
...
+        q90a = im.jpegsave_buffer(Q=90, subsample_mode=0)
```
Comment (alon-ne) [inline test/test-suite/test_foreign.py]: Let's rename q90a -> q90_subsample_auto

**Review by alon-ne — CHANGES_REQUESTED** [review]

File: libvips/foreign/vips2jpeg.c (hunk 1 above)
Comment (Elad-Laufer) [reply libvips/foreign/vips2jpeg.c]: yes, but I leave it here for backwards compatibility.

Comment (Elad-Laufer) [reply libvips/foreign/vips2jpeg.c]: simplified by setting  subsample_mode=VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF when no_subsample=1

File: test/test-suite/test_foreign.py [hunk 6]
```diff
@@ -259,14 +259,19 @@ def test_jpegsave(self):
...
-        b1 = im.jpegsave_buffer(Q=10)
-        b2 = im.jpegsave_buffer(Q=90)
-        assert len(b2) > len(b1)
+        q10 = im.jpegsave_buffer(Q=10)
+        q10_subsample_off = im.jpegsave_buffer(Q=10, subsample_mode=2)
+        q10_subsample_no = im.jpegsave_buffer(Q=10, no_subsample=1)
+        q90 = im.jpegsave_buffer(Q=90)
+        assert len(q90) > len(q10)
+        assert len(q10_subsample_off) > len(q10)
+        assert len(q10_subsample_off) == len(q10_subsample_no)
```
Comment (golanpa) [inline test/test-suite/test_foreign.py:269]: missing test:
q10_subsample_no_and_subsample_mode_on = im.jpegsave_buffer(Q=10, no_subsample=1, subsample_mode=1)

Comment (Elad-Laufer) [reply test/test-suite/test_foreign.py:269]: done

**Review by golanpa — APPROVED** [review]

**Pull request merged**